packages/astro/src/runtime/server/render/common.ts

```diff
@@ -69,7 +69,7 @@ export class HTMLParts {
 	}
 	toArrayBuffer() {
 		this.parts.forEach((part, i) => {
-			if (typeof part === 'string') {
+			if(!ArrayBuffer.isView(part)) {
 				this.parts[i] = encoder.encode(String(part));
```

Comment:
This string is a HTMLString which is not `typeof === 'string'`. Better to check if it is a typed array.